qoqo_for_braket/qoqo_for_braket/backend.py

```diff
@@ -886,3 +923,157 @@ def run_program_queued(
                 queued_runs.append(self.run_measurement_queued(measurement))
 
         return queued_runs
+
+
+def virtual_z_replacement(
+    circuit: Circuit, rotation_map: Dict[int, CalculatorFloat], apply_final_rz: bool = False
+) -> Tuple[Circuit, Optional[Dict[int, CalculatorFloat]]]:
+    """Replace the Z gate with the virtual Z gate in a quantum circuit.
+
+    It commutes the phasse at the end of the circuit.
```

Comment:
```suggestion
    It commutes the phase at the end of the circuit.
```

Could you elaborate on this, maybe with the equation for one of the gates?

Reply:
And add the explanation to the init of the backend